### PR TITLE
Migrate turtlebot-bringup to ros2 humble

### PR DESCRIPTION
**Pull request by giusebar**
Migrate turtlebot-bringup to ros2 humble

### CODE REVIEW

File: .github/workflows/snap.yaml [hunk 1]
```diff
@@ -1,12 +1,11 @@
-name: test turtlebot multi snap
-
+name: snap
 on:
   push:
     branches:
-      - feature/multi-snap
+      - giuseppe/turtlebot-bringup-humble-devel
```
Comment (MirkoFerrati) [inline .github/workflows/snap.yaml]: branch name to be fixed

File: turtlebot3c-bringup-snap/snap/local/core_launcher.sh [hunk 2]
```diff
@@ -8,8 +7,8 @@ if [[ -z "${TURTLEBOT3_MODEL}" ]]; then
...
 fi
 
-if [[ -z "${TURTLEBOT3_SIMULATION}" ]]; then
-    TURTLEBOT3_SIMULATION="$(snapctl get simulation)"
+if [[ -z "${SIMULATION}" ]]; then
+    SIMULATION="$(snapctl get simulation)"
 fi
 
-$SNAP/roslaunch turtlebot3c_bringup turtlebot3c_bringup.launch simulation:=$TURTLEBOT3_SIMULATION
+ros2 launch turtlebot3_bringup robot.launch.py use_sim_time:=$SIMULATION
```
Comment (Guillaumebeuzeboc) [inline turtlebot3c-bringup-snap/snap/local/core_launcher.sh]: This launch is going to launch the whole tb3 stack right? https://github.com/ROBOTIS-GIT/turtlebot3/blob/humble-devel/turtlebot3_bringup/launch/robot.launch.py#L88

Isn't it accessing USB and serial port?

File: turtlebot3c-bringup-snap/snap/snapcraft.yaml [hunk 3]
```diff
@@ -4,40 +4,40 @@ license: GPL-3.0
...
-    snap set turtlebot3c-bringup lds-model=LDS-02
+  By default the snap used the lidar LDS-01 robot.
+  To select the other lidar for the robot
 
-  This snap is meant to run on the Turtlebot3 robot,
-  but it can also be used with the simulation.
+    snap set turtlebot3c lds-model=LDS-02
 
-  In order to use it with the simulation, issue the following in the terminal,
+  By default the snap runs the waffle_pi Turtlebot3 model.
```
Comment (Guillaumebeuzeboc) [inline turtlebot3c-bringup-snap/snap/snapcraft.yaml]:
```suggestion
  By default the snap runs the `waffle_pi` Turtlebot3 model.
```

File: turtlebot3c-bringup-snap/README.md [hunk 4]
```diff
@@ -21,7 +21,7 @@ it only starts the `robot_state_publisher` since the robot is meant to be spawne
 ## How to install
```
Comment (Guillaumebeuzeboc) [inline turtlebot3c-bringup-snap/README.md:20]: In the instructions above it stills mention the robot_state_publisher. Is it on purpose?

File: turtlebot3c-bringup-snap/snap/local/core_launcher.sh (hunk 2 above)
Comment (giusebar) [reply turtlebot3c-bringup-snap/snap/local/core_launcher.sh]: yes, that part is gonna not work and fail and just the robot_state_publisher is gonna stay up. Do you think we should still have a higher level file that avoids launching stuff that won't work?

File: turtlebot3c-bringup-snap/README.md (hunk 4 above)
Comment (giusebar) [reply turtlebot3c-bringup-snap/README.md:20]: That is still true tho, I mean the robot_state_publisher is the only thing that will stay alive with simulation set to True

File: turtlebot3c-bringup-snap/snap/snapcraft.yaml (hunk 3 above)
Comment (giusebar) [reply turtlebot3c-bringup-snap/snap/snapcraft.yaml]: should this be changed even tho is in the snapcraft.yaml description?

Comment (Guillaumebeuzeboc) [reply turtlebot3c-bringup-snap/snap/snapcraft.yaml]: apparently, before it was:
```
 By default, the snap is configured for the 'waffle_pi' Turtlebot3 model.
```

Comment (Guillaumebeuzeboc) [reply turtlebot3c-bringup-snap/snap/snapcraft.yaml]: was it causing an issue.

Comment (giusebar) [reply turtlebot3c-bringup-snap/snap/snapcraft.yaml]: ah the description was from the monolithic snap, which didn't have the quotes. Updated with the multi-snap description for consistency.

File: turtlebot3c-bringup-snap/snap/snapcraft.yaml [hunk 5]
```diff
@@ -4,19 +4,19 @@ license: GPL-3.0
...
-    snap set turtlebot3c lds-model=LDS-02
+  Similarly, the snap is configured to consider the lidar model LDS-01.
+  To select the other lidar for the robot run the following,
 
-  By default the snap runs the waffle_pi Turtlebot3 model.
-  To select another robot model (waffle or burger)
+    snap set turtlebot3c-bringup lds-model=LDS-02
 
-    snap set turtlebot3c turtlebot3-model=burger
+  This snap is meant to run on the Turtlebot3 robot.
```
Comment (Guillaumebeuzeboc) [inline turtlebot3c-bringup-snap/snap/snapcraft.yaml]:
```suggestion
  This snap is meant to run on the TurtleBot3 robot.
```

File: turtlebot3c-bringup-snap/snap/snapcraft.yaml [hunk 6]
```diff
@@ -4,19 +4,19 @@ license: GPL-3.0
 summary: Turtlebot3c core snap
 description: |
   This snap automatically spawn a roscore and the core components for the
-  Turtlebot3.
+  Turtlebot3 (controller, robot_state_publisher etc).
 
-  This snap is meant to run on the Turtlebot3 robot.
+  By default, the snap is configured for the 'waffle_pi' Turtlebot3 model.
```
Comment (Guillaumebeuzeboc) [inline turtlebot3c-bringup-snap/snap/snapcraft.yaml]:
```suggestion
  By default, the snap is configured for the 'waffle_pi' TurtleBot3 model.
```

File: turtlebot3c-bringup-snap/snap/snapcraft.yaml [hunk 7]
```diff
@@ -4,19 +4,19 @@ license: GPL-3.0
 summary: Turtlebot3c core snap
 description: |
   This snap automatically spawn a roscore and the core components for the
-  Turtlebot3.
+  Turtlebot3 (controller, robot_state_publisher etc).
```
Comment (Guillaumebeuzeboc) [inline turtlebot3c-bringup-snap/snap/snapcraft.yaml]:
```suggestion
  TurtleBot3 (controller, robot_state_publisher etc).
```

File: turtlebot3c-bringup-snap/snap/snapcraft.yaml [hunk 8]
```diff
@@ -4,19 +4,19 @@ license: GPL-3.0
 summary: Turtlebot3c core snap
```
Comment (Guillaumebeuzeboc) [inline turtlebot3c-bringup-snap/snap/snapcraft.yaml]:
```suggestion
summary: TurtleBot3c core snap
```

**Review by Guillaumebeuzeboc — APPROVED** [review]

File: .github/workflows/snap.yaml [hunk 9]
```diff
@@ -16,195 +15,14 @@ on:
...
-            sudo snap stop turtlebot3c-nav.navigation
-
-            kill $pid
-            sudo snap stop turtlebot3c-bringup
-
-            EOF
-
-            bash test-script.sh
+      snap-name: turtlebot3c-bringup
+      publish: ${{ github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
```
Comment (MirkoFerrati) [inline .github/workflows/snap.yaml:28]: please ask for a new channel in the store and use it here as a parameter

**Review by MirkoFerrati — CHANGES_REQUESTED** [review]

File: .github/workflows/snap.yaml (hunk 9 above)
Comment (giusebar) [reply .github/workflows/snap.yaml:28]: Added the track variable. Channels created https://forum.snapcraft.io/t/track-request-turtlebot3c-bringup-noetic-humble/41035/2

**Review by MirkoFerrati — APPROVED** [review]

**Pull request merged**